Source/Core/VideoCommon/UberShaderPixel.cpp

```diff
@@ -0,0 +1,1130 @@
+            "  uint4 bpmem_pack2[8];\n"   // .x - tevorder, .y - tevksel
+            "  int4  konstLookup[32];\n"
+            "  bool  bpmem_rgba6_format;\n"
+            "  bool  bpmem_dither;\n"
+            "  bool  bpmem_bounding_box;\n"
+            "};\n\n");
+  out.Write("#define bpmem_combiners(i) (bpmem_pack1[(i)].xy)\n"
+            "#define bpmem_tevind(i) (bpmem_pack1[(i)].z)\n"
+            "#define bpmem_tevorder(i) (bpmem_pack2[(i)].x)\n"
+            "#define bpmem_tevksel(i) (bpmem_pack2[(i)].y)\n\n");
+
+  // Shader inputs/outputs in GLSL (HLSL is in main).
+  if (ApiType == APIType::OpenGL || ApiType == APIType::Vulkan)
+  {
+    if (use_dual_source)
+    {
+      if (DriverDetails::HasBug(DriverDetails::BUG_BROKEN_FRAGMENT_SHADER_INDEX_DECORATION))
+      {
+        out.Write("FRAGMENT_OUTPUT_LOCATION(0) out vec4 ocol0;\n");
+        out.Write("FRAGMENT_OUTPUT_LOCATION(1) out vec4 ocol1;\n");
+      }
+      else
+      {
+        out.Write("FRAGMENT_OUTPUT_LOCATION_INDEXED(0, 0) out vec4 ocol0;\n");
+        out.Write("FRAGMENT_OUTPUT_LOCATION_INDEXED(0, 1) out vec4 ocol1;\n");
+      }
+    }
+    else
+    {
+      out.Write("FRAGMENT_OUTPUT_LOCATION(0) out vec4 ocol0;\n");
+    }
+
+    if (!early_depth)
+      out.Write("#define depth gl_FragDepth\n");
+
+    if (g_ActiveConfig.backend_info.bSupportsGeometryShaders || ApiType == APIType::Vulkan)
+    {
+      out.Write("VARYING_LOCATION(0) in VertexData {\n");
+      GenerateVSOutputMembers(out, ApiType, numTexgen, false,
+                              GetInterpolationQualifier(msaa, ssaa));
+
+      if (g_ActiveConfig.iStereoMode > 0)
+        out.Write("  flat int layer;\n");
+
+      out.Write("};\n\n");
+    }
+    else
+    {
+      out.Write("%s in float4 colors_0;\n", GetInterpolationQualifier(msaa, ssaa));
+      out.Write("%s in float4 colors_1;\n", GetInterpolationQualifier(msaa, ssaa));
+      // compute window position if needed because binding semantic WPOS is not widely supported
+      // Let's set up attributes
+      for (u32 i = 0; i < numTexgen; ++i)
+        out.Write("%s in float3 tex%d;\n", GetInterpolationQualifier(msaa, ssaa), i);
+      out.Write("%s in float4 clipPos;\n", GetInterpolationQualifier(msaa, ssaa));
+      if (g_ActiveConfig.bEnablePixelLighting)
+      {
+        out.Write("%s in float3 Normal;\n", GetInterpolationQualifier(msaa, ssaa));
+        out.Write("%s in float3 WorldPos;\n", GetInterpolationQualifier(msaa, ssaa));
+      }
+    }
+  }
+
+  // Uniform index -> texture coordinates
+  if (numTexgen > 0)
+  {
+    if (ApiType != APIType::D3D)
+    {
+      out.Write("float3 selectTexCoord(uint index) {\n");
+    }
+    else
+    {
+      out.Write("float3 selectTexCoord(uint index");
+      for (u32 i = 0; i < numTexgen; i++)
+        out.Write(", float3 tex%u", i);
+      out.Write(") {\n");
+    }
+
+    out.Write("  switch (index) {\n");
+    for (u32 i = 0; i < numTexgen; i++)
+    {
+      out.Write("  case %u:\n"
```

Comment:
This needs a 'u suffix else Mesa complains.

`type mismatch with switch init-expression and case label (int != uint)`